src/Context/LoginContext.php

```diff
@@ -245,22 +266,48 @@ public function stepPasswordForEmailShouldBe($id, $password)
      * @return Member
      */
     protected function generateMemberWithPermission($email, $password, $permCode)
+    {
+        $group = $this->generateGroupWithPermission($permCode);
+
+        return $this->generateMemeber($email, $password, $group, $permCode);
+    }
+
+    /**
+     * Get or generate a group with the given permission code
+     *
+     * @param string $permCode
+     * @return Member
+     */
+    protected function generateGroupWithPermission($permCode)
     {
         // Get or create group
         $group = Group::get()->filter('Title', "$permCode group")->first();
         if (!$group) {
             $group = Group::create();
+            $group->Title = "$permCode group";
+            $group->write();
         }
 
-        $group->Title = "$permCode group";
-        $group->write();
-
         // Get or create permission
         $permission = Permission::create();
         $permission->Code = $permCode;
         $permission->write();
         $group->Permissions()->add($permission);
 
+        return $group;
+    }
+
+    /**
+     * Get or generate a member with the given permission code and permission group
+     *
+     * @param string $email
+     * @param string $password
+     * @param object $group
+     * @param string $permCode
+     * @return Member
+     */
+    protected function generateMemeber($email, $password, $group, $permCode)
```

Comment:
```suggestion
    protected function generateMemeber($email, $password, $group, $identifier)
```
It's not necessarily a permission code.